code/js/modules/BaseController.js

```diff
@@ -214,7 +209,12 @@
       song: this.getSongData(this.selectors.song),
       artist: this.getSongData(this.selectors.artist),
       isPlaying: this.isPlaying(),
-      siteName: this.siteName
+      siteName: this.siteName,
+      canDislike: !!(this.selectors.dislike && this.doc().querySelector(this.selectors.dislike)),
+      canPlayPrev: !!(this.selectors.playPrev && this.doc().querySelector(this.selectors.playPrev)),
+      canPlayPause: !!((this.selectors.playPause && this.doc().querySelector(this.selectors.playPause)) || (this.selectors.play && this.doc().querySelector(this.selectors.play)) || (this.selectors.pause && this.doc().querySelector(this.selectors.pause))),
```

Comment:
I have this here to accomodate future functionality which I partially coded here https://github.com/nemchik/streamkeys/commit/f850d90fd9ebe6ac2d7fae3431f4a3cc1d7f68e7#diff-177a0981b4a05d3862ef8239f6f8922aR184 and mentioned in the conversation tab of the PR. The method of thinking is that `hasPlayer` is always true if `canPlayPause` is true. So if the code from the linked commit in this comment is used in the future then `hasPlayer` would just be replaced by `canPlayPause`.